toolz/itertoolz.py

```diff
@@ -97,6 +98,16 @@ def groupby(key, seq):
     return rv
 
 
+def indices(*sizes):
+    """ Iterates over a length/shape.
+
+    >>> list(indices(3, 2))
+    [(0, 0), (0, 1), (1, 0), (1, 1), (2, 0), (2, 1)]
+    """
+
+    return(itertools.product(*[range(_) for _ in sizes]))
```

Comment:
Yep, sorry, this was kind of sloppy on my part. Will clean this up. Thanks for the tips.